xonsh/parsers/base.py

```diff
@@ -19,13 +22,55 @@
 from xonsh.tokenize import SearchPath, StringPrefix
 from xonsh.lazyasd import LazyObject
 from xonsh.parsers.context_check import check_contexts
+from xonsh.events import events
+from xonsh.tools import print_exception
```

Comment:
It's down at 375.

I didn't think shells applied to script execution, though? I would like to be able to apply it before history, so that history backends may choose to log pre- and post-transformation.